cmd/start.go

```diff
@@ -279,6 +254,45 @@ func (x *Start) Execute(args []string) error {
 		return err
 	}
 
+	if x.BitcoinCash {
+		walletCfg.Type = "bitcoincash"
+	} else if x.ZCash != "" {
+		walletCfg.Type = "zcashd"
+		walletCfg.Binary = x.ZCash
+	}
+
+	ct := wallet.Bitcoin
+	switch(walletCfg.Type) {
+	case "bitcoincash":
+		ct = wallet.BitcoinCash
+	case "zcashd":
+		ct = wallet.Zcash
+	}
+	sqliteDB, err := InitializeRepo(repoPath, x.Password, "", isTestnet, time.Now(), ct)
+	if err != nil && err != repo.ErrRepoExists {
+		return err
+	}
+
+	// If the database cannot be decrypted, exit
+	if sqliteDB.Config().IsEncrypted() {
+		sqliteDB.Close()
+		fmt.Print("Database is encrypted, enter your password: ")
+		bytePassword, _ := terminal.ReadPassword(int(syscall.Stdin))
+		fmt.Println("")
+		pw := string(bytePassword)
+		sqliteDB, err = InitializeRepo(repoPath, pw, "", isTestnet, time.Now(), ct)
+		if err != nil && err != repo.ErrRepoExists {
+			return err
+		}
+		if sqliteDB.Config().IsEncrypted() {
+			log.Error("Invalid password")
+			os.Exit(3)
+		}
+	}
+
+	// Get creation date. Ignore the error and use a default timestamp.
+	creationDate, _ := sqliteDB.Config().GetCreationDate()
```

Comment:
¯\_(ツ)_/¯ 

If it errors (or isn't set) then creationDate will just be the unix epoch. Which is fine as the wallet just uses it to decide if it can skip downloading some blocks to save time. Worst case is the sync just takes longer. Not sure if it's worth panicing here rather than just proceeding with a longer sync.

Reply:
I guess the problem is that it shouldn't return an error. Do we ever care about that value?

Reply:
I could see a scenario where I'd probably want to catch an error rather than return a default. But the start function isn't one of them.

Reply:
I would try to log that there was an error and that block sync will start from the beginning.